src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1380,7 +1380,12 @@ NativeParameterBinderController.ArgumentPassingStyle switch
             string rawPath =
                 context.EngineSessionState.GetNamespaceCurrentLocation(
                     context.ProviderNames.FileSystem).ProviderPath;
-            startInfo.WorkingDirectory = WildcardPattern.Unescape(rawPath);
+            
+            // only set this if the working directory still exists
+            if (Directory.Exists(rawPath))
+            {
+                startInfo.WorkingDirectory = WildcardPattern.Unescape(rawPath);
+            }
```

Comment:
1. With this we can still have a race condition.
It seems cmd.exe keeps CWD opened and thereby blocking it. But I don't think we need to complicate the code to track and open and close CWD in each Set-Location.

2. StartInfo.WorkingDirectory is used differently depended on UseShellExecute.
But if we always resolve full path to application (do we?) it is not problem.